packages/dcos-integration-test/extra/test_endpoints.py

```diff
@@ -76,10 +77,15 @@ def test_if_uiconfig_is_available(cluster):
 
 
 def test_if_dcos_history_service_is_up(cluster):
-    r = cluster.get('/dcos-history-service/ping')
 
-    assert r.status_code == 200
-    assert 'pong' == r.text
+    @retry(stop_max_delay=20000, wait_fixed=500)
```

Comment:
The nginx / adminrouter actually forwards / load balances to all master.mesos addresses: https://github.com/dcos/adminrouter/blob/master/nginx.master.conf#L17

so by hitting adminrouter, adminrouter sends it to one instance mor or less at random